homeassistant/components/vesync/switch.py

```diff
@@ -1,29 +1,54 @@
 """Support for VeSync switches."""
 
+from collections.abc import Callable
+from dataclasses import dataclass
 import logging
-from typing import Any
+from typing import Any, Final
 
 from pyvesync.vesyncbasedevice import VeSyncBaseDevice
-
-from homeassistant.components.switch import SwitchEntity
+from pyvesync.vesyncoutlet import VeSyncOutlet
+from pyvesync.vesyncswitch import VeSyncSwitch
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import DEV_TYPE_TO_HA, DOMAIN, VS_COORDINATOR, VS_DISCOVERY, VS_SWITCHES
+from .common import rgetattr
+from .const import DOMAIN, VS_COORDINATOR, VS_DISCOVERY, VS_SWITCHES
 from .coordinator import VeSyncDataCoordinator
 from .entity import VeSyncBaseEntity
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True, kw_only=True)
+class VeSyncSwitchEntityDescription(SwitchEntityDescription):
+    """A class that describes custom switch entities."""
+
+    is_on: Callable[[VeSyncBaseDevice], bool]
+
+
+SENSOR_DESCRIPTIONS: Final[tuple[VeSyncSwitchEntityDescription, ...]] = (
+    VeSyncSwitchEntityDescription(
+        key="device_status",
+        translation_key="on",
+        is_on=lambda device: device.device_status == "on",
+    ),
```

Comment:
Should we just use the power device class? We should either set the name to `None` (causing the entity to follow the device name, so `switch.vesync_device`) or if there already is another main feature of the device (say a climate entity or humidifier), we should use the device class translation for the POWER device class

Reply:
Any harm in us not filtering if say it is a fan or humidifer?  name would be switch.vesync_device and allow a toggle for power.  This keeps the code simple as I don't need to filter if one of those.

Reply:
Duplicate entity? Humidifier and fan both derive from ToggleEntity and expose the same actions.

Reply:
I will test this in conjunction with humidifier PR once that one is merged.   Lets plan to merge this after that one.

Reply:
Sorry but why do we need this entity?

The `device_status` in [pyvesync ](https://github.com/webdjoe/pyvesync/blob/master/src/pyvesync/vesyncfan.py) seems to reflect the on/off state which fan/humidifier entities will report as well via `is_on`. They also have turn_on/turn_off to change the status. I did not check but I think the same should be true for light entities.

Reply:
They used to sell outlets and switches.  As per: https://www.home-assistant.io/integrations/vesync/ so this would show on off for the light switch or outlet.  I am thinking I should add a filter ability so it can be removed for select situations.    In the future this switch platform can offer other items though, which is the driver for the refactor.

Reply:
Switch and outlet would need that but I don't think they should be needed for fan/humidifiers.

My humidifier appears as following with off state.
![image](https://github.com/user-attachments/assets/9cfc228f-d9a7-40a4-bb3c-b2491b404171)
![image](https://github.com/user-attachments/assets/8508a9a3-6609-4060-ba6d-0e0f176a282f)

Reply:
I will add an additional check so we can adjust for that.  I won't get to this right a way since I think humidifier merge and binary_sensor take priority.

Reply:
Added the checks to ensure this is just outlet/switch.